coral-spark/src/main/java/com/linkedin/coral/spark/IRRelToSparkRelTransformer.java

```diff
@@ -339,6 +339,12 @@ public RexNode visitCall(RexCall call) {
      */
     private Optional<RexNode> swapExtractUnionFunction(RexCall call) {
       if (call.getOperator().getName().equalsIgnoreCase("extract_union")) {
+        // Only when there's a necessity to register coalesce_struct UDF
```

Comment:
If there are multiple `extract_union` occurrences in the SQL query, the `sparkUDFInfos` list will contain multiple copies of the same `SparkUDFInfo`, will this potential redundancy cause any problems?

Reply:
Good catch, I don't think it will cause problems for Spark, but I would also prefer to check if it exists before adding it.

Reply:
I was thinking maybe define `sparkUDFInfos` to a `Set`, instead of `List`? But this seems to need to refactor some other areas of code too.

Reply:
Yeah I had the same thought before, but changing it to `Set` also needs to modify the code in Spark. Another way is to keep using `List` and deduplicate before constructing `CoralSpark`

Reply:
Resolving this conversation as there's a separate PR addressing this issue. https://github.com/linkedin/coral/pull/209/files